connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockPlayerInputTranslator.java

```diff
@@ -44,5 +52,42 @@ public void translate(PlayerInputPacket packet, GeyserSession session) {
         );
 
         session.sendDownstreamPacket(clientSteerVehiclePacket);
+
+        // Bedrock only sends movement vehicle packets while moving
+        // This allows horses to take damage while standing on magma
+        Entity vehicle = session.getRidingVehicleEntity();
+        boolean sendMovement = false;
+        if (vehicle instanceof AbstractHorseEntity && !(vehicle instanceof LlamaEntity)) {
+            sendMovement = vehicle.isOnGround();
+        } else if (vehicle instanceof BoatEntity) {
+            if (vehicle.getPassengers().size() == 1) {
+                // The player is the only rider
+                sendMovement = true;
+            } else {
+                // Check if the player is the front rider
+                Vector3f seatPos = session.getPlayerEntity().getMetadata().getVector3f(EntityData.RIDER_SEAT_POSITION);
+                if (seatPos != null && seatPos.getX() > 0) {
+                    sendMovement = true;
+                }
+            }
+        }
+        if (sendMovement) {
+            long timeSinceVehicleMove = System.currentTimeMillis() - session.getLastVehicleMoveTimestamp();
+            if (timeSinceVehicleMove >= 100) {
```

Comment:
Should you set the `lastVehicleMovementTimestamp` here?

Reply:
No, as that is set in `BedrockMoveEntityAbsoluteTranslator`. The goal of this is to send a `ClientVehicleMovePacket` with every `ClientSteerVehiclePacket`.